solutions/ks-dependent.mdx

```diff
@@ -0,0 +1,157 @@
+---
+id: ks-dependent
+source: Kick Start 2021 Round H
+title: Dependent Events 
+author: Albert Ye
+---
+
+[Official Analysis](https://codingcompetitions.withgoogle.com/kickstart/round/0000000000435914/00000000008d9970#analysis)
+
+## Implementation
+Time complexity: $\mathcal{O}(N \log N + Q \log N)$. The $\mathcal{O}(N \log N)$ comes from generating the sparse tables, and the $\mathcal{O}(Q \log N)$ comes from processing the LCA.
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+const int MOD = 1e9+7;
+
+// due to Benjamin Qi
```

Comment:
?

Reply:
yeah my bad that was from back when u could "credit" code snippets in usaco